gums-service/src/main/java/org/opensciencegrid/gums/service/JSONMapper.java

```diff
@@ -105,6 +109,49 @@ private void doVersion(HttpServletRequest request,
     }
 
 
+    private void doGenerateOsgUserVoMap(HttpServletRequest request,
+                          HttpServletResponse response)
+        throws ServletException, IOException
+    {
+        Configuration configuration = null;
+        try
+        {
+            configuration = gums.getConfiguration();
+        }
+        catch (Exception e)
+        {
+            errorResponse(response, response.SC_BAD_REQUEST, "Failed to load GUMS configuration: " + e.getMessage());
+        }
+
+        String hostname = request.getParameter("hostname");
+        if (hostname == null)
+        {
+            errorResponse(response, response.SC_BAD_REQUEST, "Missing required input parameter 'hostname'");
+            return;
+        }
+
+        String map = null;
+        try
+        {
+            map = gums.generateOsgUserVoMap(hostname);
```

Comment:
Yeah, I saw that `generateOsgUserVoMap` tests for `hasReadAllAccess(currentUser(), hostname)`, here:
https://github.com/opensciencegrid/gums/blob/v1.5.2/gums-core/src/main/java/gov/bnl/gums/admin/GUMSAPIImpl.java#L236

But if the access requirements should be done differently let me know.